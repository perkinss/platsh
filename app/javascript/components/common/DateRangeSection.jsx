import React, {useState} from "react";
import {Grid} from "@material-ui/core";
import {DatePicker, MuiPickersUtilsProvider} from "@material-ui/pickers";
import DateFnsUtils from "@date-io/date-fns";
import { createMuiTheme } from "@material-ui/core/styles";
import ThemeProvider from "@material-ui/styles/ThemeProvider";

const lightPickerTheme = createMuiTheme({
    palette: {
        primary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
});

export default function DateRangeSection(props) {

    const {fromDate, toDate, setFromDate, setToDate, callback } = props

    const handleFromDateChange = (date) => {
        setFromDate(date)
        callback()
    }
    const handleToDateChange = (date) => {
        setToDate(date)
        callback()
    }

    return (
        <Grid container spacing={2} justify={"flex-start"}>
            <Grid item xs={6} sm={6} xl={6}>
                <MuiPickersUtilsProvider  utils={DateFnsUtils}>
                    <ThemeProvider theme={lightPickerTheme}>
                        <DatePicker
                            variant={'dialog'}
                            format={'MMMM dd, yyyy'}
                            margin={'normal'}
                            fullWidth={true}
                            autoOk={true}
                            clearable={true}
                            style={{marginTop: '0'}}
                            label={"From:"}
                            value={fromDate || null}
                            onChange={handleFromDateChange}
                        />
                    </ThemeProvider>
                </MuiPickersUtilsProvider>
            </Grid>
            <Grid item xs={6} sm={6} xl={6}>
                <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <ThemeProvider theme={lightPickerTheme}>
                        <DatePicker
                            variant={'dialog'}
                            format={'MMMM dd, yyyy'}
                            margin={'normal'}
                            fullWidth={true}
                            autoOk={true}
                            clearable={true}
                            style={{marginTop: '0'}}
                            label={"Until:"}
                            value={toDate || null}
                            onChange={handleToDateChange}
                        />
                    </ThemeProvider>
                </MuiPickersUtilsProvider>
            </Grid>
        </Grid>
            )
}