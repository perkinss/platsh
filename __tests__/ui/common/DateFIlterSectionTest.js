import DateRangeSection from "../../../app/javascript/components/common/DateRangeSection";
import expect from "expect";
import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {createRender, createShallow} from "@material-ui/core/test-utils";

Enzyme.configure({ adapter: new Adapter() });

describe('<DateFilterSection />', () => {
    let shallow, render

    beforeAll(() => {
        shallow = createShallow();
        render = createRender()
    })

    it('should match the snapshot', () => {
        const wrapper = shallow(
             <DateRangeSection />
        )
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });
})