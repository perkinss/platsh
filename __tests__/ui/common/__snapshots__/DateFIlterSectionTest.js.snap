// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateFilterSection /> should match the snapshot 1`] = `ShallowWrapper {}`;
